ruby-challenges.rb

```diff
@@ -19,12 +56,46 @@
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
 # Pseudo code:
+#       input: a hash
+#       output: an array with hash values in alphabetical order
+#       process: make new hash?
+#                need to join all three arrays
+#                set in alpha order
+#                .sort
+#                google search points towards flatten method, which basically takes arrays with sub arrays joins them into one single array
+#                my nemisis is knowing how exactly to put all together, where are my corner pieces?
+
+def all_state(hash)
+end
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
-# Pseudo code:
+# Pseudo code: 
+#       input: three methods
+#       output: 'The Trek bike has 2 wheels and is going 0 mph.'
+#       process: creating the class
+#                inialize with model, wheels, mph
+#                maybe the attr_accessor instead of setting several gettor and settr methods ?
+#                string interpolation
+
+# class Bike
+#     def initialize(model, wheels, current_speed)
+#         @model = model 
+#         @wheels = wheels
+#         @current_speed = current_speed
+#     end
+
+#     def get_bike_info
+#         "The #{@model} has #{@wheels} and is going#{@current_speed}mph."
+#     end
+# end
+# bike = Bike.new('Trek', 2, 0)
```

Comment:
Remember, upon initialization the bike should have 2 wheels and a speed of 0 mph. So, instead of those instance variables of '@wheels' and '@speed' assigned to variable they should be assigned to values